src/timesync/timesyncd-manager.c

```diff
@@ -477,16 +477,16 @@ static int manager_receive_response(sd_event_source *source, int fd, uint32_t re
         m->missed_replies = 0;
 
         /* check our "time cookie" (we just stored nanoseconds in the fraction field) */
-        if (be32toh(ntpmsg.origin_time.sec) != m->trans_time.tv_sec + OFFSET_1900_1970 ||
-            be32toh(ntpmsg.origin_time.frac) != (unsigned long) m->trans_time.tv_nsec) {
+        if (be32toh(ntpmsg.origin_time.sec) != (unsigned int)(m->trans_time.tv_sec + OFFSET_1900_1970) ||
+            be32toh(ntpmsg.origin_time.frac) != (unsigned int) m->trans_time.tv_nsec) {
```

Comment:
hmm, so I figure you want to say `uint32_t` here, right?

The C standard gives a lot of freedom how integer overflows may be handled, and C compilers use that to optimize code. Thus we shouldn't rely on normal overflow here and casts, but handle the overflows explicitly, and make sure they always work the same way so that we can recognize our own packets properly.

i.e. maybe add helpers here that cast these values to uint32_t in a safe, reproducible fashion.